test/unit/org/apache/cassandra/service/accord/AccordKeyspaceTest.java

```diff
@@ -97,4 +134,157 @@ public void serde()
         Command loaded = AccordKeyspace.loadCommand(store, id);
         Assertions.assertThat(loaded).isEqualTo(committed);
     }
+
+    @Test
+    public void findOverlappingKeys()
+    {
+        var tableIdGen = fromQT(CassandraGenerators.TABLE_ID_GEN);
+        var partitionGen = fromQT(CassandraGenerators.partitioners());
+
+        var sstableFormats = DatabaseDescriptor.getSSTableFormats();
+        List<String> sstableFormatNames = new ArrayList<>(sstableFormats.keySet());
+        sstableFormatNames.sort(Comparator.naturalOrder());
+
+        List<String> memtableFormats = MemtableParams.knownDefinitions().stream()
+                                                     .filter(name -> !name.startsWith("test_") && !name.equals("default"))
+                                                     .sorted()
+                                                     .collect(Collectors.toList());
```

Comment:
I honestly added this because I found data loss once we hit SSTables... turns out internal write logic doesn't do enough validation (as we do this in CQL) so we can write to Memtable, but this gets filtered out when we write to SSTables...  Testing the different formats only came in after to help me debug this issue... and figured its nice to do this anyways =)